server/client.go

```diff
@@ -2491,47 +2492,86 @@ func (c *client) addShadowSubscriptions(acc *Account, sub *subscription) error {
 	}
 
 	var (
-		_ims   [16]ime
-		ims    = _ims[:0]
-		tokens []string
-		tsa    [32]string
-		hasWC  bool
+		_ims           [16]ime
+		ims            = _ims[:0]
+		imTsa          [32]string
+		tokens         []string
+		tsa            [32]string
+		hasWC          bool
+		tokensModified bool
 	)
 
 	acc.mu.RLock()
-	// Loop over the import subjects. We have 3 scenarios. If we have an
+	subj := string(sub.subject)
+	if len(acc.imports.streams) > 0 {
+		tokens = tsa[:0]
+		tokenizeSubjectIntoSlice(subj, &tokens)
+		for _, tk := range tokens {
+			if tk == pwcs {
+				hasWC = true
+			}
+		}
+		if tokens[len(tokens)-1] == fwcs {
+			hasWC = true
+		}
+	}
+	// Loop over the import subjects. We have 4 scenarios. If we have an
 	// exact match or a superset match we should use the from field from
-	// the import. If we are a subset, we have to dynamically calculate
-	// the subject.
+	// the import. If we are a subset or overlap, we have to dynamically calculate
+	// the subject. On overlap, ime requires the overlap subject.
 	for _, im := range acc.imports.streams {
 		if im.invalid {
 			continue
 		}
-		subj := string(sub.subject)
 		if subj == im.to {
-			ims = append(ims, ime{im, false})
+			ims = append(ims, ime{im, "", false})
```

Comment:
I know it is a bit silly, but for consistency sake, replace `""` with `_EMPTY_`.

Reply:
done